src/lib/grub.rs

```diff
@@ -33,8 +37,12 @@ pub fn get_boot_counter(grub_path: &str) -> Result<Option<i32>> {
 }
 
 /// sets grub variable boot_counter if not set
-pub fn set_boot_counter(reboot_count: u16, grub_path: &str, mount_info_path: &str) -> Result<()> {
-    match get_boot_counter(grub_path) {
+pub fn set_boot_counter(reboot_count: u16) -> Result<()> {
+    set_boot_counter_at(reboot_count, GRUB_PATH)
+}
+
+fn set_boot_counter_at(reboot_count: u16, grub_path: &str) -> Result<()> {
```

Comment:
I am not sure if the `set_boot_counter` function should fail if the boot counter is already set, that logic seems to be more appropriate for the `main` function. I mean the I would expect the grub library to set the boot counter to the value I want independently if it is already set or not.